.github/workflows/tests.yml

```diff
@@ -0,0 +1,185 @@
+name: Tests
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: npm ci
```

Comment:
This should be [gated by a cache miss](https://github.com/actions/cache#example-cache-workflow) otherwise it will always run and invalidate the cache.

```yml
      - name: Install dependencies
        if: steps.cache-deps.outputs.cache-hit != 'true'
        run: npm ci
```

Reply:
Why/how would `npm ci` invalidate a cache?

Reply:
https://docs.npmjs.com/cli/v9/commands/npm-ci

> If a node_modules is already present, it will be automatically removed before npm ci begins its install

Reply:
We're now caching `~/.npm` so this is no longer a concern.